src/jwt/issue.spec.js

```diff
@@ -0,0 +1,79 @@
+import jwt from 'jsonwebtoken';
+import lolex from 'lolex';
+import { JWT_HMAC_SECRET, JWT_RSA_PRIVATE_KEY } from './__fixtures__/keys';
+
+import { issueJWT } from './index';
+
+describe('issueJWT', () => {
+	let clock;
+	beforeEach(() => {
+		clock = lolex.install({ now: 1556030699872, toFake: ['Date'] });
+	});
+	afterEach(() => {
+		clock.uninstall();
+	});
+
+	it('should throw if unknown algorithm is passed', async () => {
+		let jwtError;
+		const wrongAlgorithm = 'test';
+		try {
+			await issueJWT('test', wrongAlgorithm, JWT_HMAC_SECRET);
+		} catch (error) {
+			jwtError = error;
+		}
+
+		expect(jwtError).toEqual(new Error(`Unknown algorithm ${wrongAlgorithm}`));
+	});
+
+	it('should generate a HMAC JWT', async () => {
+		const token = await issueJWT('test', 'HMAC', JWT_HMAC_SECRET);
+		expect(typeof token).toBe('string');
+		const decoded = jwt.decode(token, { complete: true });
+		expect(decoded).toEqual({
+			header: { alg: 'HS512', typ: 'JWT' },
+			payload: { typ: 'JWT', iat: 1556030699, exp: 1556034299, sub: 'test' },
+			signature:
+				'-1vGyim9LfmNfserg1vQHGq25bNukicO4T7wlSNtFhKwdBVy52G3eojMc4qOurcL-Jgekd72TBh-gST264_Ouw'
+		});
+	});
+
+	it('should generate a RSA JWT', async () => {
+		const token = await issueJWT('test', 'RSA', JWT_RSA_PRIVATE_KEY);
+		expect(typeof token).toBe('string');
+		const decoded = jwt.decode(token, { complete: true });
+		expect(decoded).toEqual({
+			header: { alg: 'RS512', typ: 'JWT' },
+			payload: { typ: 'JWT', iat: 1556030699, exp: 1556034299, sub: 'test' },
+			signature:
+				'OOdeGa0q_aAF6w9A5GDjH5jSnALjbnev85_PgSZiBZNEJLOQBW78ajkySqMT6sUexKHdlIzyP4pOBYAejpgTO5zDOjvYXJ_sC_DEd0kSJK5WMQcCeSmNugfBIvm4mrAAdWJjfwh9OYsVcy8I_Zr8J7kSxaoeI_L0LjR1ZqGmCoXmTZ1c5viJGDLCougPfNA6UENY6nK9UhffhPi-heC4qtHYNOhOUiMkhrm01xuOVkkjIoN2f_1MoI1ZO_mj0NZ7E1mCKRbiD_KsqNLYfxcv8OH83PB16ha5r7bN8oO7vIsCusecXD52EcFSMs0N3Bte-901kJVMsgwncWzzLd5-lw'
+		});
+	});
+
+	it('should setup typ', async () => {
+		const typ = 'test_typ';
+		const token = await issueJWT('test', 'HMAC', JWT_HMAC_SECRET, typ);
```

Comment:
Not really sure we should use the `typ` claim with the intent to differentiate between types of tokens in our applications. This is not the recommended use: https://tools.ietf.org/html/rfc7519#page-11

Reply:
OK, I will come up with a different key